### DISCUSSION TITLE
DietPi-Software | SearXNG

### DISCUSSION THREAD

**Discussion by Orko79** [MichaIng/DietPi#5508] [Software Request]
#### Formal software information
- Software title | SearXNG
- Short description | self hosted search engine
- Official URL (if available) | <https://docs.searxng.org/>
- GitHub URL (if available) | <official: https://github.com/searxng/searxng/ | docker image for ARM: https://hub.docker.com/r/searxng/searxng - GitHub https://github.com/searxng/searxng-docker or https://github.com/searxng/searxng/ without docker>
- (Official) install documentation | check Github

#### Are there similar/alternative software titles available with DietPi-Software?
no
- ...

#### What makes your requested software better than the above solutions, if available?
It is fully open source. It keeps my data in my house and will not send any private informations somewhere. I think I´m not the only one who would love to host his own, private and secure search engine 
- ...

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?
According to the official install instructions there are a lot of manual steps to do. I was not able to set up a running instance of searx, not "the normal way" and not using docker. But I´m  using Linux for just about half a year now, so the problem may sit in front of the monitor :-)
- ...

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
eehhmmm just make it work please :o) 

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.
I wold if i could! I can help with reporting issues, but I think I´m not able to do more, sorry! I´m not an experienced user so far.
- ...

**Comment by Joulinar** [comment]
I moved the request die GitHub discussions. Vote for this software implementation with the ⬆️ button on this post.

**Comment by useronegit** [comment]
searx and searxng would fit great in dietpi because it would be very easy to install through dietpi and easier to get started using it without complicated things

**Comment by hubbsterx** [comment]
for metager is down for free usage and startpage is no more private anymore, duckduckgo and co. is in the US, quant in France with also no privacy, searx seems to be the last possibility to search private! And it would be very great to have it within the dietpi-software!

**Comment by sunjam** [comment]
Update, searx has been archived as of 2023, so searxng is the only choice.  [Direct link to readme](https://github.com/searx/searx?tab=readme-ov-file#is-searx-in-maintenance-mode):

"searx is no longer maintained."

**Comment by elchmartin** [comment]
would really love to see it.
SearxNG is still supported as far as I understand it.

**Comment by giloumail** [comment]
Do you think it's possible to install searxng on Dietpi? A port to Dietpi would be really great! 

https://docs.searxng.org/

**Comment by Nutjob** [comment]
Still hoping for a porting of SearXNG on DietPi...

**Comment by mcduarte2000** [comment]
Arrived here looking at how to do it in DietPi...

Anyway, following these instructions, if you have docker installed, works:

https://docs.searxng.org/admin/installation-docker.html

(of course you need docker installed)
